infra/src/main/kotlin/fi/fta/geoviite/infra/ratko/RatkoClient.kt

```diff
@@ -41,15 +41,20 @@ class RatkoClient @Autowired constructor(private val client: WebClient) {
         jsonMapper { addModule(kotlinModule { configure(KotlinFeature.NullIsSameAsDefault, true) }) }
             .disable(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES)
 
-    fun ratkoIsOnline(): Boolean {
+    data class RatkoStatus(val statusCode: HttpStatus  ) {
+        val isOnline = statusCode == HttpStatus.OK
+    }
+
+    fun ratkoIsOnline(): RatkoStatus {
         return client
             .get()
             .uri("/api/versions/v1.0/version")
             .retrieve()
-            .onStatus({ !it.is2xxSuccessful }, { it.createException() }) //only accept 2xx as valid status code
             .toBodilessEntity()
-            .thenReturn(true)
-            .onErrorResume { Mono.just(false) }
+            .thenReturn(RatkoStatus(HttpStatus.OK))
+            .onErrorResume(WebClientResponseException::class.java) {
+                Mono.just(RatkoStatus(it.statusCode))
+            }
```

Comment:
Tuossa mietin, että pitäisikö meillä on vaan OK = 200, ClientError kaikki 400, ServerError kaikki 500 eli mäppäis esim. 403 -> 400. Sinällään operaattori ei tee tuolla httpStatus tiedolla mitään.

Reply:
Ja tähän liittyen itse asiassa, että mitä käy status codeille 300 (?)

Reply:
Periaatteessa pari vaihtoehtoa. Joko voidaan sokeasti seurata redirectiä (ei ehkä fiksua?), tai sitten näyttää virheilmo. Ehkäpä virheilmoitus kuulostaisi fiksummalta. Todennäköisesti niissä tilanteissa yhteyttä voisi pyytää ottamaan meihin eikä ratkolaisiin, koska meidänkin pitää selvittää onko esim. Ratkon osoite konffattu väärin.

Reply:
Nyt on lisätty käsittelyt 300/500-sarjojen statuskoodeille